ocaml/lambda/printlambda.ml

```diff
@@ -162,41 +162,51 @@ let tag_and_constructor_shape print_value_kind ppf (tag, shape) =
     (constructor_shape print_value_kind)
     shape
 
-let variant_kind print_value_kind ppf ~consts ~non_consts =
-  fprintf ppf "@[<hov 1>[(consts (%a))@ (non_consts (%a))]@]"
+let variant_kind or_null_suffix print_value_kind ppf ~consts ~non_consts =
+  fprintf ppf "@[<hov 1>[(consts (%a))@ (non_consts (%a))%s]@]"
     (Format.pp_print_list ~pp_sep:Format.pp_print_space Format.pp_print_int)
     consts
     (Format.pp_print_list ~pp_sep:Format.pp_print_space
       (tag_and_constructor_shape print_value_kind))
     non_consts
+    or_null_suffix
 
 let value_kind print_value_kind_non_null ppf vk =
-  match vk.nullable with
-  | Non_nullable ->
-    print_value_kind_non_null ppf vk.raw_kind
-  | Nullable ->
-    fprintf ppf "%a or_null" print_value_kind_non_null vk.raw_kind
+  let or_null_suffix =
+    match vk.nullable with
+    | Non_nullable -> ""
+    | Nullable -> " or_null"
+  in
+  print_value_kind_non_null or_null_suffix ppf vk.raw_kind
 
-let rec value_kind_non_null ppf = function
+let rec value_kind_non_null or_null_suffix ppf = function
   | Pgenval -> ()
```

Comment:
Here and in `value_kind_non_null'` we don't print `or_null_suffix` in the `Pgenval` case.

Reply:
The `Pgenval` cases are intentional: I couldn't find a good way to print it, and I was hoping that it wouldn't be very important.